pkg/clients/gitops/space_requests.go

```diff
@@ -24,3 +24,22 @@ func (g *GitopsController) GetSpaceRequests(namespace string) (*codereadytoolcha
 
 	return spaceRequestList, nil
 }
+
+// GetSpaceRequest returns spaceRequest if the name exists in the namespace
+func (g *GitopsController) GetSpaceRequest(namespace, name string) (*codereadytoolchainv1alpha1.SpaceRequest, error) {
+	spaceRequest := &codereadytoolchainv1alpha1.SpaceRequest{}
+
+	err := g.KubeRest().Get(context.Background(), client.ObjectKey{
+		Namespace: namespace,
+		Name:      name,
+	}, spaceRequest)
+
+	if err != nil {
+		if k8sErrors.IsNotFound(err) {
+			return nil, nil
+		}
+		return nil, fmt.Errorf("error occurred while trying to get SpaceRequest %s/%s: %v", namespace, name, err)
+	}
+
+	return spaceRequest, nil
```

Comment:
hmm I'm not sure if returning `nil, nil` in case the `SpaceRequest` was **not found** is safe.

would it make sense to rather do something like this:
```suggestion
	return spaceRequest, err
```